api/src/task/notifications/sync_suppressed_emails.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime, timedelta
+
+from sqlalchemy import select
+
+from src.adapters import db
+from src.adapters.aws.sesv2_adapter import BaseSESV2Client, SESV2Client
+from src.db.models.user_models import LinkExternalUser, SuppressedEmail
+from src.task.task import Task
+
+
+class SyncSuppressedEmailsTask(Task):
+    def __init__(self, db_session: db.Session, sesv2_client: BaseSESV2Client | None = None) -> None:
+        super().__init__(db_session)
+
+        self.sesv2_client = sesv2_client or SESV2Client()
+
+    def run_task(self) -> None:
+        # Get the most recent suppression timestamp from DB
+        stmt = select(SuppressedEmail).order_by(SuppressedEmail.last_update_time.desc()).limit(1)
+        with self.db_session.begin():
```

Comment:
This'll let you unindent all the code one level while still having the db_session.begin() wrapper. 

```suggestion
    def run_task(self) -> None:
        with self.db_session.begin():
            self.process_suppressed_emails()
            
   def process_suppressed_emails(self) -> None:
        # Get the most recent suppression timestamp from DB
        stmt = select(SuppressedEmail).order_by(SuppressedEmail.last_update_time.desc()).limit(1)
        ...
```